src/containers/Funds.tsx

```diff
@@ -4,19 +4,25 @@ import "./Funds.css";
 import { Table } from "react-bootstrap";
 
 import * as api from "../api/base";
+import * as token from "../api/token";
+import { RouterAppProps } from "../interfaces";
 
-export default function Funds() {
+export default function Funds(props: React.PropsWithChildren<RouterAppProps>) {
     const [funds, setFunds] = React.useState([]);
 
     React.useEffect(() => {
         fetchFunds();
     }, []);
 
     const fetchFunds = async () => {
-        const funds = await api.fetchFunds();
+        const response = await api.fetchFunds();
 
-        if (!funds.detail) {
-            setFunds(funds)
+        if (response.success) {
+            setFunds(response.funds)
+        } else if (!await token.verify()) {
+            props.history.push("/login");
+        } else {
+            alert(response.message)
```

Comment:
Need some way to redirect to login if fetch funds failed. So, if we can verify we've logged in some other error has occurred.